src/mame/motorola/mex68kecb.cpp

```diff
@@ -0,0 +1,339 @@
+	m_brg->out_f<11>().set(FUNC(mex68kecb_state::write_f11_clock)); //  150bps
+	m_brg->out_f<13>().set(FUNC(mex68kecb_state::write_f13_clock)); //  110bps
+
+	// Set up PIT and ACIAs.
+
+	PIT68230(config, m_pit, 8_MHz_XTAL / 2);
+	ACIA6850(config, m_acia1, 0);
+	ACIA6850(config, m_acia2, 0);
+
+	// Set up interrupts.
+
+	// Nothing at IRQ1
+	m_pit->timer_irq_callback().set_inputline("maincpu", M68K_IRQ_2);
+	m_pit->port_irq_callback().set_inputline("maincpu", M68K_IRQ_3);
+	// Optional 6800 peripherals at IRQ4
+	m_acia1->irq_handler().set_inputline("maincpu", M68K_IRQ_5);
+	m_acia2->irq_handler().set_inputline("maincpu", M68K_IRQ_6);
+	// ABORT Button at IRQ7
+
+	// Set up terminal RS-232.
+
+	RS232_PORT(config, m_terminal, default_rs232_devices, "terminal");
+	m_terminal->rxd_handler().set(m_acia1, FUNC(acia6850_device::write_rxd));
+	m_terminal->cts_handler().set(m_acia1, FUNC(acia6850_device::write_cts));
+	m_terminal->dcd_handler().set(m_acia1, FUNC(acia6850_device::write_dcd));
+	m_acia1->txd_handler().set(m_terminal, FUNC(rs232_port_device::write_txd));
+	m_acia1->rts_handler().set(m_terminal, FUNC(rs232_port_device::write_rts));
+
+	// Set up host RS-232.
+
+	RS232_PORT(config, m_host, default_rs232_devices, nullptr);
+	m_host->rxd_handler().set(m_acia2, FUNC(acia6850_device::write_rxd));
+	m_host->cts_handler().set(m_acia2, FUNC(acia6850_device::write_cts));
+	m_host->dcd_handler().set(m_acia2, FUNC(acia6850_device::write_dcd));
+	m_acia2->txd_handler().set(m_host, FUNC(rs232_port_device::write_txd));
+	m_acia2->rts_handler().set(m_host, FUNC(rs232_port_device::write_rts));
+}
+
+void mex68kecb_state::mem_map(address_map &map)
+{
+	map.unmap_value_high();
+	map(0x000000, 0x000007).ram().w(FUNC(mex68kecb_state::bootvect_w));       /* After first write we act as RAM */
+	map(0x000000, 0x000007).rom().r(FUNC(mex68kecb_state::bootvect_r));       /* ROM mirror just during reset */
+	map(0x000008, 0x007fff).ram(); /* 32KB RAM */
+	map(0x008000, 0x00bfff).rom().region("roms", 0); /* 16KB ROM */
+	map(0x010000, 0x01003f).rw("pit", FUNC(pit68230_device::read), FUNC(pit68230_device::write)).umask16(0x00ff);
+	map(0x010040, 0x010043).rw("acia1", FUNC(acia6850_device::read), FUNC(acia6850_device::write)).umask16(0xff00);
+	map(0x010040, 0x010043).rw("acia2", FUNC(acia6850_device::read), FUNC(acia6850_device::write)).umask16(0x00ff);
+}
+
+void mex68kecb_state::machine_start ()
+{
+	/* Setup pointer to bootvector in ROM for bootvector handler bootvect_r */
+	m_sysrom = (uint16_t *)(memregion("roms")->base());
+}
+
+void mex68kecb_state::machine_reset ()
+{
+	// Reset BRG.
+	m_brg->rsa_w(CLEAR_LINE);
+	m_brg->rsb_w(ASSERT_LINE);
+
+	/* Reset pointer to bootvector in ROM for bootvector handler bootvect_r */
+	if (m_sysrom == &m_sysram[0]) /* Condition needed because memory map is not setup first time */
+		m_sysrom = (uint16_t*)(memregion("roms")->base());
+}
+
+void mex68kecb_state::write_f1_clock(int state)
```

Comment:
This would be much cleaner as a template function, with the bit number as the template parameter.

Reply:
Guess I'll figure out how to do that! (This is the first C++ I've written since 2005...)

Reply:
I never ask people to do something I haven't already done myself :-)

Here's an example of a function that works that way: https://github.com/mamedev/mame/blob/049d10067fbfe79099c79860268b705f5f880ffc/src/mame/apple/iosb.cpp#L324

Reply:
The function itself is fairly straightforward.  The devilish detail is pre-instantiating all of the versions you're going to use, which you can see immediately after the function there at line 356.